upload_form.py

```diff
@@ -272,6 +269,16 @@ def set_columns(filename, fileformat=None):
     else:
         add_is_gal_if_needed(table, True)
 
+    # Rename the uploaded file to something unique, and store this name in the table
+    extension = os.path.splitext(filename)[-1]
+    full_filename_old = os.path.join(app.config['UPLOAD_FOLDER'], filename)
+    file = open(full_filename_old)
+    unique_filename = hashlib.sha1(file.read()).hexdigest()[0:36-len(extension)] + extension
+    file.close() 
```

Comment:
A nicer way to do this is:
```
with open(full_filename_old) as file:
    unique_filename = hashlib.sha1(file.read()).hexdigest()[0:36-len(extension)] + extension
```
because it takes care of closing automatically